src/web/mjs/connectors/templates/SixParkbbs.mjs

```diff
@@ -22,53 +23,46 @@ export default class SixParkbbs extends Connector {
         let data = await this.fetchDOM(request, this.queryMangaTitle);
-        let id = this.getRootRelativeOrAbsoluteLink(uri, request.url);
+        let id = this.getRootRelativeOrAbsoluteLink(uri, this.url);
         let title = data[0].textContent.replace(/^\s*\[.*?\]\s*/g, '').trim();
         return new Manga(this, id, title);
     }
 
     async _getMangas() {
         let mangaList = [];
-        let request = new Request(new URL(this.path.replace('%PAGE%', 1000), this.url), this.requestOptions);
+        let request = new Request(new URL(this.sub + this.path.replace('%PAGE%', 1000), this.url), this.requestOptions);
         let data = await this.fetchDOM(request, this.queryMangasPageCount);
         let pageCount = parseInt(data[0].href.match(this.pathMatch)[1]);
         for(let page = 1; page <= pageCount; page++) {
             let mangas = await this._getMangasFromPage(page);
             mangaList.push(...mangas);
         }
-        return mangaList.filter(ele=> {
-            return ele != undefined;
-        });
+        return mangaList;
     }
 
     async _getMangasFromPage(page) {
-        let request = new Request(new URL(this.path.replace('%PAGE%', `${page}`), this.url), this.requestOptions);
+        let request = new Request(new URL(this.sub + this.path.replace('%PAGE%', `${page}`), this.url), this.requestOptions);
         let data = await this.fetchDOM(request, this.queryMangas);
-        return data.map(element => {
-            if(element.text.match(this.queryMangasMatch)){
-                return {
-                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
-                    title: element.text.trim()
-                };
-            }
-            return;
+        return data.filter(element=> {
+            return element.text.match(this.queryMangasMatch) != null;
+        }).map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, (this.url + this.sub)),
+                title: element.text.trim()
+            };
         });
     }
 
     async _getChapters(manga) {
-        let chapterList = [null]
-        chapterList = chapterList.map(element =>{
-            return{
-                id: manga.id,
-                title: manga.title
-            }
-        });
-        return chapterList;
+        return [{
+            id: manga.id,
+            title: manga.title
+        }];
     }
 
     async _getPages(chapter) {
-        let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        let request = new Request(new URL(this.sub + chapter.id, this.url), this.requestOptions);
         let data = await this.fetchDOM(request, this.queryPage);
-        return data.map(element => this.getAbsolutePath(element.attributes[0].value, request.url));
+        return data.map(element => this.getAbsolutePath(element.attributes.mydatasrc.value || element.attributes.src.value, request.url));
```

Comment:
I added `|| element.attributes.src.value` because some manga do not have `element.attributes.mydatasrc.value` but still getting `Cannot read property 'value' of undefined`, any idea where i did wrong?

Reply:
Try `element.getAttribute('mydatasrc') || element`, the `getAbsolutePath` automatically detect if it is an image with the `src` attribute and use it if available

Reply:
nice, this work